packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -40,228 +34,106 @@ def ensure_routable(cmd, host, port):
-            app['portDefinitions'] = [{'labels': {'VIP_0': vip}}]
+def vip_app(container: Container, network: Network, host: str, vip: str, offset_port: bool):
+    """ If offset_port is True then the port on the user network will be one higher
+    to allow two apps on the same USER network
+    """
+    if network in [Network.HOST, Network.BRIDGE]:
+        # both of these cases will rely on marathon to assign ports
+        return get_test_app(
+            network=network,
+            host_constraint=host,
+            vip=vip,
+            container_type=container)
+    elif network == Network.USER:
+        # ports must be incremented as one shared network is used
+        user_port = 8000 if not offset_port else 8001
+        return get_test_app(
+            network=network,
+            host_port=user_port,
+            host_constraint=host,
+            vip=vip,
+            container_type=container)
     else:
-        app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
-                     '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py 9080'
-        app['container']['docker']['portMappings'] = [{
-            'hostPort': 0,
-            'containerPort': 9080,
-            'protocol': 'tcp',
-            'name': 'test',
-            'labels': {}
-        }]
-        if vip is not None:
-            app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': vip}
-        if network == 'USER':
-            app['ipAddress'] = {'networkName': 'dcos'}
-    app['constraints'] = [['hostname', 'CLUSTER', host]]
-    return app, uuid
-
-
-def mesos_vip_app(num, network, host, vip, ucr=False):
-    app = None
-    uuid = None
-    port = backend_port_st + num
-    if ucr is False:
-        app, uuid = get_test_app()
+        raise AssertionError('Unexpected network: {}'.format(network.value))
+
+
+def generate_vip_app_permutations():
+    """ Generate all possible network interface permutations for applying vips
+    """
+    network_options = [Network.USER, Network.BRIDGE, Network.HOST]
+    permutations = []
+    for container in [Container.NONE, Container.MESOS, Container.DOCKER]:
+        for named_vip in [True, False]:
+            for same_host in [True, False]:
+                for vip_net in network_options:
+                    for proxy_net in network_options:
+                        if container != Container.DOCKER and Network.BRIDGE in (vip_net, proxy_net):
+                            # only DOCKER containers support BRIDGE network
+                            continue
+                        permutations.append((container, named_vip, same_host, vip_net, proxy_net))
+    return permutations
+
+
+@pytest.mark.slow
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
+@pytest.mark.parametrize('container,named_vip,same_host,vip_net,proxy_net', generate_vip_app_permutations())
+def test_vip(dcos_api_session, container, named_vip, same_host, vip_net, proxy_net):
```

Comment:
This is an excellent candidate to to follow the documentation guidelines in code-style guide. The content is already present, please format it according to the style guidelines.